### DISCUSSION TITLE
Will this plugin work after migration from sfdx to sf (v2) ?

### DISCUSSION THREAD

**Discussion by pavanvemunuri** [scolladon/sfdx-git-delta#686] [General]
Hi @scolladon,
We are currently using sfdx cli and sfdx-git-delta plugin in our Jenkins pipeline.
Can you confirm if this plugin still works with sf (v2) after migrating from sfdx ?

Thanks in advance.

**Comment by scolladon** [comment]
Hi @pavanvemunuri,

We currently do not know, we have no tests related to `sf` v2.

We are planning to work on migrating the plugin to `sf` v2 on september.

**Reply by pavanvemunuri** [reply]
Thanks for the quick update. Will wait for the announcement about sf v2 upgrade.

**Comment by mehdicherf** [comment]
Also, @pavanvemunuri, please let us know if you face any issue with the current version of SGD  while testing your pipeline with sf (v2).

**Reply by pavanvemunuri** [reply]
We are planning to wait till migration of the SGD plugin to sf (v2) . Will let you know in case if we try with current version of SGD.

**Comment by scolladon** [comment]
Hi @pavanvemunuri 

We have tested the plugin with `sf` v2 and it works perfectly fine.
To ensure that over time we have included a e2e test in our pipeline dedicated to test the plugin with `sf` v2 (installation, execution, e2e test).

However, as the plugin architecture has not yet been migrated to the `sf` v2 plugin one, the command still requires to be called using `:`.
- `sf sgd:source:delta <params>` works fine
- `sfdx sgd:source:delta <params>` works fine too (using the `sfdx` alias to `sf`)
- `sf sgd source delta <params>` does not work _yet_

We plan to migrate to the `sf` v2 plugin architecture in the next months and will think of a way to provide a smooth transition.
It will fix the [warning message](https://github.com/scolladon/sfdx-git-delta/issues/619) about `apiVersion`.

Until that you can use `sfdx` or `sf`, both works

**Reply by pavanvemunuri** [reply]
Hi @scolladon ,
Thanks a lot for the update. We will start upgrading to `sf` v2. 
Will keep an eye on your migration updates. Your quick support is much appreciated!

**Reply by pavanvemunuri** [reply]
Hi @scolladon ,
Hope you are doing well.
We did start setting up sf (v2) and installed sfdx-git-delta. It is currently throwing errors "sfdx: command not found" and "sf: command not found" for the below commands respectively.
Can you please let me know if anything has changed in your plugin recently causing this ?

`sfdx sgd:source:delta --from $from --to $to --output target --generate-delta`

`sf sgd:source:delta --from $from --to $to --output target --generate-delta`

**Reply by scolladon** [reply]
We have not changed anything recently related to this @pavanvemunuri.
In our test the plugin is still compatible with `sf` or `sfdx` for `mac` or `linux` or `windows` and with node `v18` or `v20`.

What is the environment you experience this behavior ?
Have you tried uninstall and reinstall `sfdx cli` ?

**Reply by pavanvemunuri** [reply]
Hi @scolladon,
Thanks for your response. Let me give some context of our setup.
We use containers in our Jenkins pipeline so that these tools are installed in each build that runs.
For your reference, we install sf v2 as below in each build and followed by sfdx-git-delta plugin installation. Our node version is 18.
Also including the console output as we received some warnings during the plugin installation and states that "_These warnings can only be addressed by the owner(s) of sfdx-git-delta._". Can you please let me know if this is something to look at.

**Installation snippet for your reference:**

```
sfTar=sf-linux-x64.tar.gz
sfLibDir=$HOME/local/lib/sf
# Installing Salesforce CLI...
curl --silent \
     --show-error \
     --output ${sfTar} \
     https://developer.salesforce.com/media/salesforce-cli/sf/channels/stable/${sfTar}
mkdir -pv ${sfLibDir}
tar --extract --gzip --file ${sfTar} --directory ${sfLibDir} --strip-components 1
rm -fv ${sfTar}

# Check install
echo "sf: $(${sfLibDir}/bin/sf version)"
export PATH=$PATH:${sfLibDir}/bin
# Installing SFDX-Git-Delta plugin...
echo y | sfdx plugins:install sfdx-git-delta
```




**Sample log output:**
```
node version: v18.17.1
Installing Salesforce CLI...
mkdir: created directory '/home/jenkins/local'
mkdir: created directory '/home/jenkins/local/lib'
mkdir: created directory '/home/jenkins/local/lib/sf'
removed 'sf-linux-x64.tar.gz'
You acknowledge and agree that the CLI tool may collect usage information, user environment, and crash reports for the purposes of providing services or functions that are relevant to use of the CLI tool and product improvements.

sf: @salesforce/cli/2.23.20 linux-x64 node-v20.10.0
Installing SFDX-Git-Delta plugin...
? This plugin isn't signed by Salesforce. Only install the plugin if you trust 
its creator. Do you want to continue the installation?, (y/N) [G? This plugin isn't signed by Salesforce. Only install the plugin if you trust 
its creator. Do you want to continue the installation?, (y/N) y[G? This plugin isn't signed by Salesforce. Only install the plugin if you trust 
its creator. Do you want to continue the installation?, Yes
Because you approved this plugin, you can avoid future installation confirmations by adding the plugin to the unsignedPluginAllowList.json file. For details, see https://developer.salesforce.com/docs/atlas.en-us.sfdx_setup.meta/sfdx_setup/sfdx_setup_allowlist.htm.
Finished digital signature check.
Installing plugin sfdx-git-delta@latest... yarn add v1.22.21
Installing plugin sfdx-git-delta@latest... info No lockfile found.Installing plugin sfdx-git-delta@latest... [1/4] Resolving packages...Installing plugin sfdx-git-delta@latest... [2/4] Fetching packages...Installing plugin sfdx-git-delta@latest... [3/4] Linking dependencies...Installing plugin sfdx-git-delta@latest... [4/4] Building fresh packages...Installing plugin sfdx-git-delta@latest... success Saved lockfile.Installing plugin sfdx-git-delta@latest... success Saved 235 new dependencies.Installing plugin sfdx-git-delta@latest... info Direct dependenciesInstalling plugin sfdx-git-delta@latest... └─ sfdx-git-delta@5.33.0Installing plugin sfdx-git-delta@latest... info All dependenciesInstalling plugin sfdx-git-delta@latest... ├─ @babel/runtime-corejs3@7.23.8Installing plugin sfdx-git-delta@latest... ├─ @oclif/screen@3.0.8Installing plugin sfdx-git-delta@latest... ├─ abort-controller@3.0.0Installing plugin sfdx-git-delta@latest... ├─ argparse@1.0.10Installing plugin sfdx-git-delta@latest... ├─ commander@4.1.1Installing plugin sfdx-git-delta@latest... ├─ dir-glob@3.0.1Installing plugin sfdx-git-delta@latest... ├─ fast-deep-equal@3.1.3Installing plugin sfdx-git-delta@latest... ├─ fs-extra@11.2.0Installing plugin sfdx-git-delta@latest... ├─ ieee754@1.2.1Installing plugin sfdx-git-delta@latest... ├─ is-number@7.0.0Installing plugin sfdx-git-delta@latest... ├─ jsonwebtoken@9.0.0Installing plugin sfdx-git-delta@latest... ├─ minimatch@3.1.2Installing plugin sfdx-git-delta@latest... ├─ nock@13.4.0Installing plugin sfdx-git-delta@latest... ├─ path-parse@1.0.7Installing plugin sfdx-git-delta@latest... ├─ rechoir@0.6.2Installing plugin sfdx-git-delta@latest... ├─ run-parallel@1.2.0Installing plugin sfdx-git-delta@latest... ├─ shebang-regex@3.0.0Installing plugin sfdx-git-delta@latest... ├─ string_decoder@1.3.0Installing plugin sfdx-git-delta@latest... ├─ tough-cookie@4.1.3Installing plugin sfdx-git-delta@latest... ├─ websocket-extensions@0.1.4Installing plugin sfdx-git-delta@latest... ├─ yn@3.1.1Installing plugin sfdx-git-delta@latest... └─ zip-stream@4.1.1Installing plugin sfdx-git-delta@latest... Done in 26.31s.Installing plugin sfdx-git-delta@latest... Refreshing user plugins...Installing plugin sfdx-git-delta@latest... installed v5.33.0
 ›   Warning: sfdx-git-delta > @salesforce/command@5.3.9: Package no longer 
 ›   supported. Contact Support at https://www.npmjs.com/support for more info.
 ›   Warning: "sfdx-git-delta > @salesforce/command > @oclif/test > @oclif/core
 ›    > ts-node@10.9.2" has unmet peer dependency "@types/node@*".
 ›   Warning: "sfdx-git-delta > @salesforce/command > @oclif/test > @oclif/core
 ›    > ts-node@10.9.2" has unmet peer dependency "typescript@>=2.7".

These warnings can only be addressed by the owner(s) of sfdx-git-delta.
We suggest that you create an issue at https://github.com/scolladon/sfdx-git-delta/issues and ask the plugin owners to address them.
```

**Reply by scolladon** [reply]
HI @pavanvemunuri, we also have those warnings on our [CI](https://github.com/scolladon/sfdx-git-delta/actions/runs/7505306556/job/20434415834).
But it does not fail the installation neither our e2e tests.

We are aware of this situation (here is the [issue](https://github.com/scolladon/sfdx-git-delta/issues/619) to track this) and we spiked on what we should do to migrate our plugin to the new `sf` plugin architecture. We don't know yet when we will do this work because the new `sf` plugin architecture is not compatible with the old one. And we still have a huge user base using the old `sfdx`.
We need to find a way to migrate smoothly for every users.

**Reply by pavanvemunuri** [reply]
Hi @scolladon , Thanks for the updates.
We prefer to pause upgrading to sf v2. Will wait for your plugin migration.

**Comment by bob-laz** [comment]
Hello, just wanted to say that we really appreciate this plugin and it is working well for us with sf CLI v2. I was just curious about the progress of migrating the plugin to use v2 architecture and if there was any ETA on a release for that yet?

**Reply by scolladon** [reply]
Hi @bob-laz, thanks for raising this point.

We spiked a bit on this and we are pretty clear on how we should do it.
We are not sure yet how to address it with npm so it does not break others that have not yet migrated their pipeline to sf cli v2.

So far no ETA and no branch to test.
This is tracked with this [issue](https://github.com/scolladon/sfdx-git-delta/issues/619)
You can follow it if you want to track our progress

**Reply by pavanvemunuri** [reply]
Hi @scolladon, I'd like to check if there is any ETA available now for migrating the plugin to v2 architecture.
FYI - Tried checking the updates on the above mentioned thread for the same.

**Reply by scolladon** [reply]
Hi @pavanvemunuri, there is no ETA yet.
The plugin works with both sfdx and salesforce/cli (we have CI jobs to ensure that) so there is no rush.
As we still have customer using the old cli we cannot migrate without scratch for them yet
